### PR TITLE
Throw error in case of Boolean options starting with 'no-'

### LINKED ISSUE #1000: boolean option is ignored if starts with 'no-'
I have yeoman generator that should accept options like this:
```js
    this.option('no-yarn', {
      desc: 'Do not use yarn for installing node dependencies',
      type: Boolean,
      default: false
    });
```

However, even if generator started with `yo mygen --no-yarn`, option is completely ignored:
```js
  initialising: function () {
   // prints: no yarn : false
    this.log(`no yarn : ${this.options['no-yarn']}`);
   }
```

But if I change name of option to smth like 'noyarn' it works as expected.

```
$ yo --version
1.8.5
``` 
```
yeoman-generator@0.24.1
```

### PR DESCRIPTION
**Pull request by milindl**
Fixes #1000

### CODE REVIEW

File: test/base.js [hunk 1]
```diff
@@ -701,6 +701,18 @@ describe('Base', function () {
...
+
+    it('disallows Boolean options starting with no-', function () {
+      var generator = new this.Dummy([], {
+        env: this.env,
+        resolved: 'test'
+      });
+      var addWrongOp = function() {
+        generator.option('no-op');
+      };
+      assert.throws(addWrongOp, Error, 'Does not throw Error');
```
Comment (SBoudrias) [inline test/base.js]: Don't add a third argument to assertions. These usually make the error less clear than the default messages.

File: lib/index.js [hunk 2]
```diff
@@ -209,6 +209,12 @@ Base.prototype.option = function option(name, config) {
     hide: false
   });
 
+  // Check whether boolean option is invalid (starts with no-)
+  if (config.type === Boolean && name.match(/^no-.+/)) {
+    return this.emit('error', new Error('Boolean option ' + chalk.bold(name) +
```
Comment (SBoudrias) [inline lib/index.js]: I think we can be more helpful here, maybe

```
Option name {name} cannot start with no-
To allow user to pass `--no-{name}`, use `this.option({name}, {type: Boolean})` instead.
```

**Comment by milindl** [comment]
Hi, I think I've mostly fixed the review-related changes :smile:

**Comment by SBoudrias** [comment]
Thanks for sending this out! I changed the error message a bit, so I merged the PR manually. It's on master now <3

Thanks a lot for helping us out!!

**Pull request closed without merging**